src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts

```diff
@@ -333,195 +218,123 @@ export class ReadingsGraphComponent implements OnDestroy {
-            ['0.8', 'rgba(101,201,96,1)'],
-            ['0.9', 'rgba(101,201,96,1)'],
-            ['1', 'rgba(253,231,37,1)']],
-          colorbar: {
-            tick0: 0,
-            dtick: 5
-          }
-        },
-      ],
-      layout: {
-        title: this.assetCode,
-        showlegend: true,
-        autoSize: true,
-        margin: {
-          b: 40,
-          l: 60,
-          r: 10,
-          t: 25
-        }
-      },
-      config: {
-        displayModeBar: false
-      }
+  public zoomGraph(seconds: number) {
+    const maxDataPoints = 600;
+    const bucket = seconds / maxDataPoints;
+    const length = seconds;
+    console.log(' Bucket = ', bucket, ' length = ', length);
+    this.payload = {
+      assetCode: this.assetCode,
+      start: 0,
+      len: length,
+      bucketSize: bucket
     };
-    this.generate3Dgraph();
+    this.getAssetReadings(this.payload);
   }
 
-  public async generate3Dgraph() {
-    // Initilization of DOM element to render graph
-    // takes time at fist so need some time to wait here.
-    const intervalId = setInterval(() => {
-      if (this.Graph) {
-        Plotly.newPlot(
-          this.Graph.nativeElement,
-          this.polyGraphData);
-        clearInterval(intervalId);
-      }
-    }, 100);
-  }
+  calculateGraphData(event: any) {
+    if (event['xaxis.range[0]'] === undefined) {
+      return;
+    }
 
-  public isNumber(val) {
-    return typeof val === 'number';
-  }
+    // console.log('payload', this.payload);
+    this.panning = true;
+    this.isAlive = false;
+    const maxDataPoints = 600;
+    const panClickTime = moment(event['xaxis.range[0]']).utc();
+    const panReleaseTime = moment(event['xaxis.range[1]']).utc();
 
-  selectTab(id: number, showSpinner = true) {
-    this.showSpinner = showSpinner;
-    this.selectedTab = id;
-    if (this.graphRefreshInterval === -1 && this.selectedTab === 4) {
-      this.showAssetReadingsSummary(this.assetCode, this.limit, this.optedTime);
-    } else if (this.graphRefreshInterval === -1) {
-      this.plotReadingsGraph(this.assetCode, this.limit, this.optedTime);
-    }
-  }
+    console.log('Pan Click Time ', event['xaxis.range[0]']);
+    console.log('Pan Release Time ', event['xaxis.range[1]']);
 
-  showSummaryTab() {
-    return this.numberTypeReadingsList.length > 0;
-  }
+    console.log('panClickTime utc', panClickTime.format());
+    console.log('panReleaseTime utc', panReleaseTime.format());
 
-  isEmptyObject(obj) {
-    return (obj && (Object.keys(obj).length === 0));
-  }
+    // console.log('panClickTime(unix timestamp)', moment(panClickTime.format()).valueOf() / 1000);
+    // console.log('panReleaseTime(unix timestamp)', moment(panReleaseTime.format()).valueOf() / 1000);
+
+    const panDeltaTime = moment.duration(panReleaseTime.diff(panClickTime)).asSeconds();
+    console.log('panDeltaTime', panDeltaTime);
+
+    const now = moment.utc(new Date()).valueOf() / 1000.0; // in milliseconds
```

Comment:
with `/ 1000.0` these are now in seconds. i.e. standard epoch time (with ms level precision)